env/file_system_tracer.h

```diff
@@ -150,6 +152,7 @@ class FSSequentialFileTracingWrapper : public FSSequentialFileWrapper {
  private:
   std::shared_ptr<IOTracer> io_tracer_;
   Env* env_;
+  std::string file_name_;
```

Comment:
Instead of saving the entire file name here and in the other wrapper classes, which could be pretty long as it would contain the entire path, we can consider parsing the file name and just storing the number and type. Then when writing the trace, reconstruct the file name. Its not necessary to write the full path.